File: Archive#00-001-100/082-Remove Duplicates from Sorted List II/leetcode082.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        dummy = ListNode(0)
        slow = dummy
        flag = False
        while head and head.next:
            if head.val != head.next.val:
                if not flag:
                    slow.next = head
                    slow = slow.next
                flag = False
            else:
                flag = True
            head = head.next
        slow.next = None if flag else head
        return dummy.next

File: Archive#00-001-100/082-Remove Duplicates from Sorted List II/leetcode082.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val)
#         @val = val
#         @next = nil
#     end
# end

# @param {ListNode} head
# @return {ListNode}
def delete_duplicates(head)
  dummy = ListNode.new(0)
  fast, slow = head, dummy
  slow.next = fast
  while (fast != nil) do
    while (fast.next != nil) && (fast.val == fast.next.val) do
      fast = fast.next
    end
    if (slow.next != fast) then
      slow.next = fast.next
    else
      slow = slow.next
    end
    fast = fast.next
  end
  dummy.next
end